src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -528,15 +528,15 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
                 int i = 0;
                 for (; i < parameters.Length; ++i)
                 {
-                    argumentsBuilder.Add(operationFactory.CreateArgumentOperation(ArgumentKind.Explicit, parameters[i], arguments[i]));
+                    argumentsBuilder.Add(operationFactory.CreateArgumentOperation(ArgumentKind.Explicit, parameters[i], arguments[i], arguments[i].WasCompilerGenerated));
```

Comment:
Are we always passing `arguments[i], arguments[i].WasCompilerGenerated`? If so, consider having `CreateArgumentOperation` determine `IsImplicit` from the `argument` parameter rather than requiring an addition `bool` parameter.